server/src/main.rs

```diff
@@ -155,10 +172,68 @@ fn load_config() -> Config {
     }
 }
 
+fn create_level(config: &Arc<Config>) -> LevelData {
+    let seed = get_seed(config);
+    let world_name = &config.world.name;
+    info!("Creating world '{}' with seed {}", world_name, seed);
+
+    // TODO: Generate spawn position properly
+    LevelData {
+        allow_commands: false,
+        border_center_x: 0.0,
+        border_center_z: 0.0,
+        border_damage_per_block: 0.0,
+        border_safe_zone: 0.0,
+        border_size: 0.0,
+        clear_weather_time: 0,
+        data_version: 0,
+        day_time: 0,
+        difficulty: 0,
+        difficulty_locked: 0,
+        game_type: 0,
+        hardcore: false,
+        initialized: false,
+        last_played: 0,
+        raining: false,
+        rain_time: 0,
+        seed,
+        spawn_x: 0,
+        spawn_y: 100,
+        spawn_z: 0,
+        thundering: false,
+        thunder_time: 0,
+        time: 0,
+        version: Default::default(),
+        generator_name: config.world.generator.to_string(),
+        generator_options: None,
+    }
+}
+
+fn get_seed(config: &Arc<Config>) -> i64 {
```

Comment:
Rather than passing a reference to an `Arc`, you can just pass `&Config`—`Arc` dereferences to the wrapped value.